std/format.d

```diff
@@ -62,6 +62,7 @@ import std.exception;
 import std.meta;
 import std.range.primitives;
 import std.traits;
+import std.math : FloatingPointControl;
```

Comment:
Perhaps this top-level import is the cause.

Would it be possible to move it to a local import, by splitting the definition of the `printFloat` function into an eponymous template?

Reply:
I'll take care. 

Probably I'll use a local enum for rounding modes instead of the solution with eponymous templates. That has the advantage, that I can add the fifth mode that IEEE allows for printing floats, which `FloatingPointControl` does not know about. (I plan a PR to make formatting floating point numbers pure by using this fifth mode for format and add a separate (nonpure) function, probably formatIEEE for rounding according to the processor state. I will need this enum there anyway.)

Shall I base this PR on `stable`, as it's some sort of regression?

Reply:
[Reading the floating point rounding mode is explicitly allowed in `pure` code:](https://dlang.org/spec/function#pure-functions)

>As a concession to practicality, a pure function can also:
>* read and write the floating point exception flags
>* read and write the floating point mode flags, as long as those flags are restored to their initial state upon function entry
>* perform impure operations in statements that are in a _ConditionalStatement_ controlled by a _DebugCondition_.